### PR TITLE
feat(compiler): ollama ai provider support

### PR DESCRIPTION
**Pull request by The-Best-Codes**
Addresses #851.

### CODE REVIEW

File: packages/cli/src/cli/localizer/explicit.ts [hunk 1]
```diff
@@ -51,18 +52,28 @@ export default function createExplicitLocalizer(
...
 
 function createAiSdkLocalizer(params: {
-  factory: (params: { apiKey: string; baseUrl?: string }) => LanguageModel;
+  factory: (params: { apiKey?: string; baseUrl?: string }) => LanguageModel;
   id: NonNullable<I18nConfig["provider"]>["id"];
   prompt: string;
-  apiKeyName: string;
+  apiKeyName?: string;
   baseUrl?: string;
+  authRequired?: boolean;
```
Comment (maxprilutskiy) [inline packages/cli/src/cli/localizer/explicit.ts]: let's invert and do `doNotAuth?: boolean` instead: it's more conventional, and it'd be harded to make mistakes for someone who's not familiar with the codebase v much

**Comment by The-Best-Codes** [comment]
There's an issue. You have to use colons (`:`) in Ollama model names to specify variants, but the code is splitting by provider:model and only taking the first chunk after the colon.
So for example, `ollama:qwen:0.6b` is parsed as `ollama` and `qwen`, when it should be `ollama` and `qwen:0.6b`.

**Comment by The-Best-Codes** [comment]
Not yet ready to merge. Need to update error messages for providers that don't require auth.

**Comment by The-Best-Codes** [comment]
Just found another bug or edge case where when using Ollama, after translating, it starts over again. I don't have the resources to test this efficiently, so I might need someone else's help.

**Comment by The-Best-Codes** [comment]
Ready to merge, if you don't have any feedback!

File: packages/cli/src/cli/localizer/explicit.ts [hunk 2]
```diff
@@ -51,35 +52,44 @@ export default function createExplicitLocalizer(
...
   id: NonNullable<I18nConfig["provider"]>["id"];
   prompt: string;
-  apiKeyName: string;
+  apiKeyName?: string;
   baseUrl?: string;
+  skipAuth?: boolean;
 }): ILocalizer {
-  const apiKey = process.env[params.apiKeyName];
-  if (!apiKey) {
+  const skipAuth = params.skipAuth == true;
```
Comment (The-Best-Codes) [inline packages/cli/src/cli/localizer/explicit.ts]: Should be `===` probably.

File: packages/compiler/src/lib/lcp/api/provider-details.ts [hunk 3]
```diff
@@ -22,4 +22,11 @@ export const providerDetails: Record<
     getKeyLink: "https://ai.google.dev/",
     docsLink: "https://ai.google.dev/gemini-api/docs/troubleshooting",
   },
+  ollama: {
+    name: "Ollama",
+    apiKeyEnvVar: undefined, // Ollama doesn't use env vars for keys
```
Comment (The-Best-Codes) [inline packages/compiler/src/lib/lcp/api/provider-details.ts]: Need to fix the wording of these comments. Ollama doesn't require API keys period.

**Review by maxprilutskiy — APPROVED** [review]
